projects/questionset-editor-library/src/lib/components/question/question.component.ts

```diff
@@ -1007,7 +1007,9 @@ export class QuestionComponent implements OnInit, AfterViewInit, OnDestroy {
     this.editorService.data = {};
     this.editorService.selectedSection = selectedUnitId;
     let metaData = this.getQuestionMetadata();
-    metaData.eval = activeNode?.data.metadata?.eval || data?.data?.eval;
+    if(activeNode?.data.metadata?.evalMode || data?.data?.evalMode || data?.data?.metadata?.evalMode){
```

Comment:
I think this can be refactored:
Can you try below code for it -

```
const evalMode = activeNode?.data?.metadata?.evalMode || data?.data?.evalMode || data?.data?.metadata?.evalMode;

if (evalMode) {
  metaData.evalMode = evalMode;
}
```
And and you tell me why we are checking `evalMode` in all these place

Reply:
done